### PR TITLE
tests: check that host settings like hostname are settable on core

### PR DESCRIPTION
**Pull request by mvo5**
It was reported that the writable-path setup on UC20 was incorrect.
This commit adds an explicit test for some of the settings that
can be changed on a Ubuntu Core system and ensures those work.

### CODE REVIEW

File: tests/core/system-settings/task.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+summary: System settings work
+
+prepare: |
+    cp -a /etc/hostname /etc/hostname.bak
```
Comment (cmatsuoka) [inline tests/core/system-settings/task.yaml:4]: The backup file can't be created on /etc because it's not writable.

**Comment by cmatsuoka** [comment]
The test is now correct, however it will fail on because it's trying to create a backup file on /etc which is not writable.

**Comment by cmatsuoka** [comment]
Pushed a small fix that writes the backup file elsewhere, and dereferences the symbolic link when creating the backup file (otherwise we would copy the link itself).

**Comment by cmatsuoka** [comment]
UC18 passed and UC20 failed as expected (because `/etc/writable` is double mounted there. I think the test is good but my review is tainted, I pushed a commit.

File: tests/core/system-settings/task.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+prepare: |
+    cp -aL /etc/hostname /tmp/hostname.bak
+
+restore: |
+    cp -a /tmp/hostname.bak /etc/hostname
+
+execute: |
+    echo "Check that setting the hostname works"
+    echo coole-kiste > /etc/hostname
```
Comment (anonymouse64) [inline tests/core/system-settings/task.yaml:11]: shouldn't we also check that the `hostname` command works and matches too?

**Comment by mvo5** [comment]
Fwiw, I don't think this will land as is, but it's a good test for the fix in core20 that is being worked on (or the initrd?)

File: tests/core/system-settings/task.yaml (hunk 2 above)
Comment (cmatsuoka) [reply tests/core/system-settings/task.yaml:11]: The `hostname` command doesn't care about the configuration file, it just sets and reads the hostname to/from the kernel, using  `sethostname(2)` and `gethostname(2)` (we would be testing the syscalls in this case -- we can do it, but it's orthogonal to the writable filesystem being mounted or not). IIRC in the old days we initialized the hostname using something like `hostname $(cat /etc/hostname)`, but currently it's probably a systemd thing.

**Comment by cmatsuoka** [comment]
Since we're checking the writable mount on `/etc`, we'll probably want to test the rest of the files stored there as well.

File: tests/core/system-settings/task.yaml (hunk 2 above)
Comment (bboozzoo) [reply tests/core/system-settings/task.yaml:11]: We should probably try `hostnamectl set-hostname` too.

Comment (zyga) [reply tests/core/system-settings/task.yaml:11]: This is also indirectly tested via `tests/main/hostname-control`.

File: tests/core/system-settings/task.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+execute: |
+    echo "Check that setting the hostname works"
+    hostnamectl set-hostname coole-kiste
+    MATCH coole-kiste < /etc/hostname
+    hostname | MATCH coole-kiste
+
+    echo "Check that setting the timezone works"
+    timedatectl set-timezone "Europe/Berlin"
+    date +"%Z" | MATCH CEST
```
Comment (bboozzoo) [inline tests/core/system-settings/task.yaml:19]: Otherwise it will fail half of the year.
```suggestion
    date +"%Z" | MATCH 'CE[S]?T'
```

**Comment by bboozzoo** [comment]
Hmm on core:
```
2020-06-05T07:27:12.8100039Z + timedatectl
2020-06-05T07:27:12.8100251Z + MATCH Europe/Berlin
2020-06-05T07:27:12.8100465Z /bin/bash: line 76: etc/timezone: No such file or directory
```
timedatectl does not set up /etc/timezone?

**Pull request merged**